meta-resin-common/recipes-support/resinhup/resinhup/run-resinhup.sh

```diff
@@ -249,6 +253,31 @@ function version_gt() {
     test "$(echo "$@" | tr " " "\n" | sort -V | head -n 1)" != "$1"
 }
 
+function cachedpull() {
+    local _image="$1"
+    local _image_escaped=$(echo "$_image" | sed -r "s/[\/\.:-]+/_/g")
```

Comment:
I think that on Linux only the "/" and "\0" characters are illegal in a filename, why replace the other ones? Also consecutive special characters are replaced by just one underscore (".:-" -> "_"), is that on purpose?